hapi/hedera-protobufs/services/basic_types.proto

```diff
@@ -1837,3 +1837,47 @@ message ConsensusTokenFeeScheduleAllowance {
      */
     uint64 amount_per_message = 5;
 }
+
+/**
+ * A unique, composite, identifier for a topic allowance.
+ *
+ * Each topic allowance SHALL be uniquely identified by a TopicAllowanceId.
+ * A TopicAllowanceId SHALL be recorded when created and MUST be provided in any transaction
+ * that would modify that topic allowance (such as a `consensusApproveAllowance`).
+ */
+message TopicAllowanceId {
+    /**
+     * The token that the allowance pertains to.
+     */
+    TokenID tokenId = 1;
```

Comment:
Style for field names is lower-snake-case (i.e. `token_id`) (the Java translation ends up the same).
This applies to all messages added for this PR.